doc/source/basics.rst

```diff
@@ -65,14 +65,28 @@ Note, **these attributes can be safely assigned to**!
    df.columns = [x.lower() for x in df.columns]
    df
 
-To get the actual data inside a data structure, one need only access the
-**values** property:
+Pandas objects (:class:`Index`, :class:`Series`, :class:`DataFrame`) can be
+thought of as containers for arrays, which hold the actual data and do the
+actual computation. For many types, the underlying array is a
+:class:`numpy.ndarray`. However, pandas and 3rd party libraries may *extend*
+NumPy's type system to add support for custom arrays
+(see :ref:`basics.dtypes`).
+
+To get the actual data inside a :class:`Index` or :class:`Series`, use
+the **array** property
 
 .. ipython:: python
 
-    s.values
-    df.values
-    wp.values
+   s.array
+   s.index.array
+
```

Comment:
I would also mention `to_numpy` here in context of the Series/Index, and not only for DataFrame below. Something like:

```
The `array` attribute can return a numpy array or a :class:`~pandas.api.extension.ExtensionArray`
depending on the data type of the Series or Index. In case you want to ensure a numpy array,
:func:`DataFrame.to_numpy` can be used:

.. ipython:: python

    s.to_numpy()
    np.asarray(s)

```